### PR TITLE
ledgerblue: 0.1.44 -> 0.1.47, fix build

### PR DESCRIPTION
**Pull request by RaghavSood**
###### Description of changes

Aims to address https://github.com/NixOS/nixpkgs/issues/226162

###### Things done

ndeflib: init at 0.3.3 (new dependency for ledgerblue)
nfcpy: init at 1.0.4 (new dependency for ledgerblue)
ledgerblue: 0.1.44 -> 0.1.47

ndeflib has two broken tests - as far as I can tell, this is likely due to the library not being updated for a few years and drifting

I'm unable to get tests for nfcpy working - the openssl wrapping seems somewhat flaky and fragile, and doesn't mesh well with nixpkgs.

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by RaghavSood** [comment]
Result of `nixpkgs-review pr 229241` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>ledger_agent (python310Packages.ledger_agent)</li>
    <li>python310Packages.ledgerblue</li>
    <li>python310Packages.ndeflib</li>
    <li>python310Packages.nfcpy</li>
    <li>python311Packages.ledger_agent</li>
    <li>python311Packages.ledgerblue</li>
    <li>python311Packages.ndeflib</li>
    <li>python311Packages.nfcpy</li>
  </ul>
</details>

**Comment by RaghavSood** [comment]
Result of `nixpkgs-review pr 229241` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>ledger_agent (python310Packages.ledger_agent)</li>
    <li>python310Packages.ledgerblue</li>
    <li>python310Packages.ndeflib</li>
    <li>python310Packages.nfcpy</li>
    <li>python311Packages.ledger_agent</li>
    <li>python311Packages.ledgerblue</li>
    <li>python311Packages.ndeflib</li>
    <li>python311Packages.nfcpy</li>
  </ul>
</details>

File: pkgs/development/python-modules/ndeflib/default.nix [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  format = "setuptools";
+
+  disabled = pythonOlder "3.5";
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-HVaChVi5sW8oIqQFGCQ0Y0e2at9TIOqGBwdItvJFSog=";
+  };
+
+  doTest = true;
```
Comment (wegank) [inline pkgs/development/python-modules/ndeflib/default.nix:20]:
```suggestion
  doCheck = true;
```

File: pkgs/development/python-modules/ndeflib/default.nix [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+  pythonImportsCheck = [
+    "ndef"
+  ];
+
+  nativeCheckInputs = [ pytest ];
+
+  # Two tests in AlternativeCarrierRecord fail, upstream hasn't been updated in a few years
+  checkPhase = ''
```
Comment (wegank) [inline pkgs/development/python-modules/ndeflib/default.nix:29]: Please do this in `pytestFlagsArray`.

**Comment by RaghavSood** [comment]
Superceded by #229882

**Pull request closed without merging**